src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -0,0 +1,243 @@
+    private static final String CONTENT_REM = AUDIT + "contentRemoval";
+    private static final String METADATA_MOD = AUDIT + "metadataModification";
+
+    private static final String CONTENT_ADD = EVENT_TYPE + "ing";
+    private static final String OBJECT_ADD = EVENT_TYPE + "cre";
+    private static final String OBJECT_REM = EVENT_TYPE + "del";
+
+    private static final String PREMIS_TIME = "premis:hasEventDateTime";
+    private static final String PREMIS_OBJ = "premis:hasEventRelatedObject";
+    private static final String PREMIS_AGENT = "premis:hasEventRelatedAgent";
+    private static final String PREMIS_TYPE = "premis:hasEventType";
+
+    private static final String RDF_TYPE = "rdf:type";
+
+    private static final String HAS_CONTENT = REPOSITORY + "hasContent";
+    private static final String LAST_MODIFIED = REPOSITORY + "lastModified";
+    private static final String LAST_MODIFIED_BY = REPOSITORY + "lastModifiedBy";
+    private static final String NODE_ADDED = REPOSITORY + "NODE_ADDED";
+    private static final String NODE_REMOVED = REPOSITORY + "NODE_REMOVED";
+    private static final String PROPERTY_CHANGED = REPOSITORY + "PROPERTY_CHANGED";
+
+    /**
+     * Creates a node for the audit event under the configured container.
+     *
+     * @param event
+     * @throws RepositoryException
+     */
+    public void createAuditNode(final FedoraEvent event) throws RepositoryException {
+        final String uuid = pidMinter.mintPid();
+        final JcrTools jcrTools = new JcrTools();
+        final String identifier = event.getIdentifier();
+        final String userData = event.getUserData();
+        final JsonObject json = new JsonParser().parse(userData).getAsJsonObject();
+        final String userAgent = json.get("userAgent").getAsString();
+        final Long timestamp =  event.getDate();
+        final DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        df.setTimeZone(TimeZone.getTimeZone("UTC"));
+        final String eventDate = df.format(new Date(timestamp));
+        final String userID = event.getUserID();
+        final String eventType = getEventURIs(event.getTypes());
+        final String properties = Joiner.on(',').join(event.getProperties());
+        final Node auditNode = containerService.findOrCreate(session, "/audit/" + uuid).getNode();
+
+        LOGGER.info("Audit node {} created for event.", uuid);
+
+        auditNode.addMixin("fedora:Resource");
+        auditNode.setProperty(RDF_TYPE, INTERNAL_EVENT);
+        auditNode.setProperty(RDF_TYPE, PREMIS_EVENT);
+        auditNode.setProperty(RDF_TYPE, PROV_EVENT);
+        auditNode.setProperty(PREMIS_TIME, eventDate);
+        auditNode.setProperty(PREMIS_OBJ, identifier);
+        auditNode.setProperty(PREMIS_AGENT, userID);
+        auditNode.setProperty(PREMIS_AGENT, userAgent);
+
+        // mapping event type/properties to audit event type
+        if (eventType.contains(NODE_ADDED)) {
```

Comment:
Unsurprisingly, there's some duplicated code between this and AuditSparqlProcessor.  Maybe this could be extracted to a method (getAuditEventType(String eventType, String properties)) and called from AuditSparqlProcessor?